civis/io/_tables.py

```diff
@@ -97,19 +99,22 @@ def read_civis(table, database, columns=None, use_pandas=False,
     """
     if use_pandas and NO_PANDAS:
         raise ImportError("use_pandas is True but pandas is not installed.")
+    if archive:
+        warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
+                      "Use `hidden` instead.", FutureWarning)
```

Comment:
My preference would be to leave it in both places. The first warning will be clear. The second may be a bit confusing, but that seems like less confusion than not having a warning from the calling function.